Assets/BossRoom/Scripts/Server/Net/ServerSessionManager.cs

```diff
@@ -0,0 +1,272 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using MLAPI;
+using UnityEngine;
+using MLAPI.SceneManagement;
+
+namespace BossRoom.Server
+{
+    public struct SessionPlayerData
+    {
+        public ulong ClientID;
+        public string ClientGUID;
+        public string PlayerName;
+        public Vector3 PlayerPosition;
+        public Vector3 PlayerRotation;
+        public bool IsConnected;
+
+        public SessionPlayerData(ulong clientID, string clientGUID, string name, Vector3 position, Vector3 rotation, bool isConnected = false)
+        {
+            ClientID = clientID;
+            ClientGUID = clientGUID;
+            PlayerName = name;
+            PlayerPosition = position;
```

Comment:
tested by trying to connect client with host, then having the client disconnect and reconnect.
class was the right one (instead of the default tank), but position was reset to spawn point.
not sure about the name since we seem to have a bug on our hud preventing the display of the name :(